src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -195,7 +197,7 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
                 var exceptions = new List<Exception>();
 
                 // Execute the hosting startup assemblies
-                foreach (var assemblyName in _options.HostingStartupAssemblies)
+                foreach (var assemblyName in _options.HostingStartupAssemblies.Distinct(StringComparer.OrdinalIgnoreCase))
```

Comment:
The alternative is saving this information somewhere and pass it to WebHost and log a warning there. We have a similar pattern with hosting startup assembly errors. Is it worth it?